master/internal/api_project_intg_test.go

```diff
@@ -583,6 +583,58 @@ func TestPatchProject(t *testing.T) {
 	require.Equal(t, strings.ToUpper(newKey), project.Key)
 }
 
+func TestPatchProjectRecordRedirect(t *testing.T) {
+	api, curUser, ctx := setupAPITest(t, nil)
+	wresp, werr := api.PostWorkspace(ctx, &apiv1.PostWorkspaceRequest{Name: uuid.New().String()})
+	require.NoError(t, werr)
+
+	projectName := "test-project" + uuid.New().String()
+	resp, err := api.PostProject(ctx, &apiv1.PostProjectRequest{
+		Name: projectName, WorkspaceId: wresp.Workspace.Id,
+	})
+	require.NoError(t, err)
+
+	exp := createTestExpWithProjectID(t, api, curUser, int(resp.Project.Id))
+	task := &model.Task{TaskType: model.TaskTypeTrial, TaskID: model.NewTaskID()}
+	require.NoError(t, db.AddTask(ctx, task))
+	require.NoError(t, db.AddTrial(ctx, &model.Trial{
+		State:        model.CompletedState,
+		ExperimentID: exp.ID,
+		StartTime:    time.Now(),
+	}, task.TaskID))
+
+	newName := uuid.New().String()
+	newDescription := uuid.New().String()
+	newKey := uuid.New().String()[:project.MaxProjectKeyLength]
```

Comment:
nit (non-blocking): If we every expand the max length of a project key this could cause this test to fail if the requested key includes the UUID dashes. We have a random package will generate a random string of length N from a given keyset (defaults to all-caps alphanumeric).